packages/contracts-bedrock/contracts/test/CrossDomainOwnable3.t.sol

```diff
@@ -0,0 +1,225 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Bytes32AddressLib } from "@rari-capital/solmate/src/utils/Bytes32AddressLib.sol";
+import { CommonTest, Messenger_Initializer } from "./CommonTest.t.sol";
+import { CrossDomainOwnable3 } from "../L2/CrossDomainOwnable3.sol";
+import { AddressAliasHelper } from "../vendor/AddressAliasHelper.sol";
+import { Hashing } from "../libraries/Hashing.sol";
+import { Encoding } from "../libraries/Encoding.sol";
+
+contract XDomainSetter3 is CrossDomainOwnable3 {
+    uint256 public value;
+
+    function set(uint256 _value) external onlyOwner {
+        value = _value;
+    }
+}
+
+contract CrossDomainOwnable3_Test is Messenger_Initializer {
+    XDomainSetter3 setter;
+
+    /**
+     * @notice OpenZeppelin Ownable.sol transferOwnership event
+     */
+    event OwnershipTransferred(address indexed previousOwner, address indexed newOwner);
+
+    /**
+     * @notice CrossDomainOwnable3.sol transferOwnership event
+     */
+    event OwnershipTransferred(
+        address indexed previousOwner,
+        address indexed newOwner,
+        bool isLocal
+    );
+
+    function setUp() public override {
+        super.setUp();
+        vm.prank(alice);
+        setter = new XDomainSetter3();
+    }
+
+    function test_constructor_succeeds() public {
+        assertEq(setter.owner(), alice);
+        assertEq(setter.isLocal(), true);
```

Comment:
Should it be possible to deploy the contract with a non-local initial owner?

Reply:
If we want to we could have it setup so that `_isLocal` is a constructor parameter, but it doesn't make much sense since we'd have to do a cross-domain which is one of the reasons we created `CrossDomainOwnable3.sol` in the first place.